lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,205 @@
+import glob
+import os
+from pathlib import Path
+
+import pytest
+
+from pbench.common.logger import get_pbench_logger
+from pbench.server.filetree import FileTree, TarballUnpackError
+from pbench.server.unpack_tarballs import Unpack
+
+
+@pytest.fixture()
+def make_logger(server_config):
+    """
+    Construct a Pbench Logger object
+    """
+    return get_pbench_logger("TEST", server_config)
+
+
+class TestUnpackTarballs:
+    tarlist = ["/mock/ABC/TO-UNPACK/A.tar.xz", "/mock/ABC/TO-UNPACK/B.tar.xz"]
+
+    def test_tarball_symlinks(self, monkeypatch, server_config, make_logger, caplog):
+        """Tests for an incorrect symlink which results in an unresolved location"""
+
+        def mock_resolve(self: Path, strict: bool = False):
+            """
+            Return a "full" fake file name
+
+            Args:
+                strict: Not used in the mock
+
+            Returns:
+                new Path with fake resolution
+            """
+            raise FileNotFoundError
+
+        monkeypatch.setattr(glob, "iglob", lambda x: self.tarlist)
+        monkeypatch.setattr(Path, "resolve", mock_resolve)
+        obj = Unpack(server_config, make_logger)
+        result = obj.unpack()
+        for tar in self.tarlist:
+            assert (
+                f"Tarball link, '{tar}', does not resolve to a real location"
+                in caplog.text
+            )
+        assert result.ntotal == 2
+        assert result.nsuccess == 0
+
+    def test_tarball_file(self, monkeypatch, server_config, make_logger, caplog):
+        """Show that when tarball path cannot be accessed it throws a
+        FileNotFoundError Exception"""
```

Comment:
The docstring is misleading: you're actually testing problems with the MD5 file, not the tarball path!